.devcontainer/configure.py

```diff
@@ -0,0 +1,158 @@
+      output = e.output
+  else:
+    output = subprocess.check_output(cmd, stderr=stderr)
+  return output.decode('UTF-8').strip()
+
+
+def is_nvidia_gpu_available():
+  """This function verifies NVIDIA Docker runtime is installed and 
+  available. It also checks NVIDIA GPU are available and the driver
+  is installed."""
+
+  docker_executable = which('docker')
+  if docker_executable is None:
+    raise RuntimeError("Docker is not installed or in the user path.")
+
+  command_str = f"{docker_executable} info --format '{{{{json .}}}}'"
+  data = json.loads(run_shell(command_str))
+
+  if "nvidia" not in data["Runtimes"].keys():
+    return False, "NVIDIA Docker Runtime is not available"
+
+  nvidia_smi_executable = which('nvidia-smi')
+  if nvidia_smi_executable is None:
+    return False, "NVIDIA Driver is not installed or not in the user path."
+
+  command_str = f"{nvidia_smi_executable} --query-gpu=gpu_name --format=csv"
+
+  if run_shell(command_str).splitlines()[1:]:  # Skip header
+    return True, None
+  else:
+    return False, "No NVIDIA GPU was found on this system."
+
+
+def get_input(question, default_answer='', accepted_answers=None):
+  accepted_answers = [x.strip().lower() for x in accepted_answers]
+
+  while True:
+
+    try:
+      answer = input(f"{question} ")  # pylint: disable=bad-builtin
+    except EOFError:
+      answer = default_answer
+    answer = (answer or default_answer).strip().lower()
+
+    if answer not in accepted_answers:
+      print(
+          f"Unsupported answer received: {answer}. Expected: {accepted_answers}"
+      )
+      continue
+
+    break
+
+  return answer.strip().lower()
+
+
+if __name__ == "__main__":
+
+  if os.path.isfile(os.path.join(CURRENT_DIR, "docker-compose.yml")):
+    if get_input(
+        "A `docker-compose.yml` already exists. Are you certain you want to overwrite it [y/N]?",
+        default_answer="n",
+        accepted_answers=["y", "n"]) == "n":
+      sys.exit(1)
+
+  with open(os.path.join(CURRENT_DIR, "docker-compose.base.yml")) as f:
+    docker_config = json.load(f)
+
+  use_official_img = get_input(
+      "Do you wish to use the official prebuild development containers [Y/n]?",
+      default_answer="y",
+      accepted_answers=["y", "n"]) == "y"
+
+  use_bleeding = get_input(
+      "Do you wish to use the bleeding-edge container (might be unstable) [Y/n]?",
+      default_answer="y",
+      accepted_answers=["y", "n"]) == "y"
+
+  use_nvidia_gpu = False
+  nvidia_gpu_available, err_msg = is_nvidia_gpu_available()
+  if nvidia_gpu_available:
+    use_nvidia_gpu = get_input("Do you wish to use NVIDIA GPU [y/N]?",
+                               default_answer="n",
+                               accepted_answers=["y", "n"]) == "y"
+  else:
+    print(f"NVIDIA GPUs are not available for use: {err_msg}")
+
+  docker_img_shortname = DOCKER_IMAGE_SHORTNAME_DICT[
+      "nvidia" if use_nvidia_gpu else "base"][
+          "latest" if use_bleeding else "stable"]
```

Comment:
Windows _nearly_ works with this using the Linux Docker containers. Definitely not blocking this PR, but we could also add Windows Docker containers to be used by our Windows CI and a Windows dev container.

Steps I followed:

* Install Docker Desktop https://www.docker.com/products/docker-desktop/
* Enable hardware virtualization in BIOS
* Install / set up WSL
* Remove any Windows/MSVC-specific VSCode/CMake settings from user/folder/workspace settings

  On my system, I had these:
  ```json
    "cmake.cmakePath": "C:\\Program Files\\CMake\\bin\\cmake.exe",
    "cmake.buildEnvironment": {
      "VCToolsInstallDir": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\Preview\\VC\\Tools\\MSVC\\14.31.31103\\",
      "UniversalCRTSdkDir": "C:\\Program Files (x86)\\Windows Kits\\10\\",
      "UCRTVersion": "10.0.19041.0"
    },
  ```
* Clean `build/` directory (or create a new build directory for the Linux container
* Pick a build type (e.g. `RelWithDebInfo`) and kit (e.g. `Clang 17.0.0 x86_64-pc-linux-gnu`), try to configure CMake.
  * Observe issues in LLVM's [`GetHostTriple.cmake`](https://github.com/llvm/llvm-project/blob/d8472ac238dfd2f3bae61a56903de5ef5df6360a/llvm/cmake/modules/GetHostTriple.cmake#L45-L58) and [`config.guess`](https://github.com/llvm/llvm-project/blob/main/llvm/cmake/config.guess) stemming from line endings (I have `git config core.autocrlf` set to true to use Windows-style line endings locally but change back to Unix-style line endings on commit/push: [git docs](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings)). I manually changed the line extensions in `config.guess` and was able to proceed